pkg/skaffold/deploy/helm.go

```diff
@@ -135,8 +135,8 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r v1alp
 	for k, v := range params {
 		setOpts = append(setOpts, "--set")
 		if r.ImageStrategy.HelmImageConfig.HelmConventionConfig != nil {
-			tagSplit := strings.Split(v.Tag, ":")
-			imageRepositoryTag := fmt.Sprintf("%s.repository=%s,%s.tag=%s", k, tagSplit[0], k, tagSplit[1])
+			tagIndex := strings.LastIndex(v.Tag, ":")
```

Comment:
Would it make sense to parse the image tag with something like `docker.ParseReference()`? If it works, it would handle all the edge cases better than any string parsing

Reply:
Sounds good, I wasn't aware of `docker.ParseReference()`, it's annoying that it only gives you the baseName and not also the tag.

Code updated to use `docker.ParseReference()` which leads to much nicer code.

I followed the style of the `helm_test.go` to fix up the test, but I'm not super happy about it because the test uses the SAME logic as the code it's testing.

Reply:
Doh, I realise that `docker.ParseReference()` is in a `skaffold` package, so probably makes more sense for me to update `docker.ParseReference()` to include the Tag.